docs/rules/require-default-props.md

```diff
@@ -279,9 +289,88 @@ MyStatelessComponent.propTypes = {
 };
 ```
 
+### `functions`
+
+- "defaultProps": Use `.defaultProps`.
+- "defaultArguments": Use object default arguments
+- "ignore": Ignore this rule for functional components.
+
+"defaultArguments" is recommended. We will change it to deafult later.
```

Comment:
see above: it's not, and i don't plan to ever change the default. `.defaultProps` will remain the superior approach for the foreseeable future.